### PR TITLE
Feedback v2

### PR DESCRIPTION
**Pull request by maxweldsouza**
This PR makes the following modifications to the feedback component:
- The component now preserves the last focused element when it is opened. Most of the focus management code has been removed.
- The emoji selection is now a radio group and behaves as such. The emoji can be navigated using the arrow keys and tabbing through the fields doesn't make you tab through every emoji option.
This was accomplished using Adobe's [react-aria](https://react-spectrum.adobe.com/react-aria/) which provides hooks to build accessible components.
The library is light weight and we can only import what we need.
- The emoji svgs are bundled together and we no longer need to hit external urls.
- The button component now forwards refs. This is a breaking change https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers.
- Buttons have a focus ring.
- Click outside has been implemented using [react-use](https://github.com/streamich/react-use) which lets us only import hooks that we need.
- Props for input fields can be overridden.
- Tests have been added using testing-library.
- The isomorphic-fetch dependency has been removed and we now rely on native fetch.
- The branch is updated with latest master.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
     "registry": "https://registry.npmjs.org/"
   },
   "dependencies": {
+    "@testing-library/user-event": "^13.2.0",
```
Comment (esoubihe) [inline package.json:69]: Might want to move this to dev?

File: src/Button/index.js [hunk 2]
```diff
@@ -24,6 +24,10 @@ const Button = ({
   component,
   block,
   large,
+  // Types have been used for styling so we need another prop to specify submit buttons
```
Comment (esoubihe) [inline src/Button/index.js:27]: I've been meaning to introduce a breaking change here to use `variants` like other components. Thoughts?

Comment (maxweldsouza) [reply src/Button/index.js:27]: Sounds good. I'll switch to variants. I'll also add a forwardRef to Button since that will be a breaking change too.

Comment (maxweldsouza) [reply src/Button/index.js:27]: Just noticed that this is done is master. Will update the branch.

**Comment by esoubihe** [comment]
> * The emoji selection is now a radio group and behaves as such. The emoji can be navigated using the arrow keys and tabbing through the fields doesn't make you tab through every emoji option.
>   This was accomplished using Adobe's [react-aria](https://react-spectrum.adobe.com/react-aria/) which provides hooks to build accessible components.

Thanks for looking into accessibility. We might want to consider [this](https://www.radix-ui.com/docs/primitives/components/radio-group) in the future for all radio elements.

> * The emoji svgs are bundled together and we no longer need to hit external URLs.

👌 ✨ 

This looks very good, @maxweldsouza. I do have a few requests to make this more flexible.

1. We might want to decouple the button from the component and allow setting a different triggering element here. An example of this would be the [DropdownMenu](http://localhost:52749/?path=/docs/dropdownmenu--default-story) component. This allows for the component to be used in different settings.
2. Instead of having the `enableEmail` prop, we should allow passing an email through a prop and when that's done, hide the email input field. This is useful for logged-in users. So, let's say we take an `email` prop and hide the field if it exists or if it's `false`. Thoughts?
3. Increasing the width of the feedback window by ~15%/20% would add some needed whitespace there.

Thanks

**Comment by maxweldsouza** [comment]
I have made the changes. I couldn't decouple the component in the same way as the DropdownMenu. I'm using function as a child. instead. Let me know if this works.

**Comment by esoubihe** [comment]
Thanks, it looks good. Can you please improve the docs in storybook? After that we're good to merge.

**Comment by esoubihe** [comment]
> 2\. Instead of having the `enableEmail` prop, we should allow passing an email through a prop and when that's done, hide the email input field. This is useful for logged-in users. So, let's say we take an `email` prop and hide the field if it exists or if it's `false`. Thoughts?

@maxweldsouza were you able to address this?

**Comment by maxweldsouza** [comment]
> > 2. Instead of having the `enableEmail` prop, we should allow passing an email through a prop and when that's done, hide the email input field. This is useful for logged-in users. So, let's say we take an `email` prop and hide the field if it exists or if it's `false`. Thoughts?
> 
> @maxweldsouza were you able to address this?

Yes. The component now takes an email prop and does not show the email field if it is provided. Have added an additional case to the documentation as well.

**Pull request merged**